Add ShellCheck to Travis build

jidicula opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied the relevant labels to this PR
- [ ] I've added relevant tests for my contribution
- [ ] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

<!--- For the title, please observe the following rules:
	- Provide a concise and self-descriptive title
	- Do not include the applicable issue number in the title, do it in the PR body
	- If the PR is not ready for review, convert it to a draft.
-->

## Description
<!-- describe what the PR is about. Explain the approach and possible drawbacks.It's ok to repeat some text from the related issue. -->
**Why this change was necessary**
Incorporating ShellCheck into our integration testing can help enforce
good shell scripting practices in our project.

**What this change does**
Adds ShellCheck into Travis build script.

**Any side-effects?**
Contributors should now run ShellCheck locally on their Bash scripts
before pushing them to GitHub to help ensure that the scripts are
robust and that the Travis build passes.

## Linked issues
<!-- If the PR fixes any issues, indicate it here with issue-closing keywords: e.g. Resolves #XX, Fixes #XX, Addresses #XX. Note that if you want multiple issues to be autoclosed on PR merge, you must use the issue-closing verb before each relevant issue: e.g. Resolves #1, Resolves #2 -->

Resolves #139
Related to #92 

**Additional context/notes/links**
https://github.com/koalaman/shellcheck#in-your-build-or-test-suites

ShellCheck installation instructions: https://github.com/koalaman/shellcheck#installing

po09i commented:
The website suddenly stopped being generated for no apparent reason. The code fails when executing this [line](https://github.com/shimming-toolbox/shimming-toolbox/blob/af27879add99bda7b5065d4f283b533edf2e95cb/generate-docs.sh#L33) Reading through the documentation revealed potential solutions. The pip installer suggests to be > v9.0.0 and we only have V8.1.1 on Tristano. Is it possible to upgrade it?

jcohenadad commented:
as per [this related issue](https://github.com/pypa/virtualenv/issues/1876), pip upgrade would be advisable

po09i commented:
PR is ready for review, virtualenv was downloaded using pip3 on tristano by agah, removing the call to download using pip fixes the problem.

jidicula committed:
Add ShellCheck to Travis build

**Why this change was necessary**
Incorporating ShellCheck into our integration testing can help enforce
good shell scripting practices in our project.

**What this change does**
Adds ShellCheck into Travis build script.

**Any side-effects?**
Contributors should now run ShellCheck locally on their Bash scripts
before pushing them to GitHub to help ensure that the scripts are
robust.

**Additional context/notes/links**

Resolves #139
Related to:

PR #92

https://github.com/koalaman/shellcheck#in-your-build-or-test-suites

ShellCheck installation instructions: https://github.com/koalaman/shellcheck#installing

jidicula committed:
Use safer shellcheck command for Travis build

**Why this change was necessary**
The previous command fails if no Bash scripts exist at the specified
path.

**What this change does**
Checks if any shellscripts exist at the specified path and only runs
ShellCheck if they do.

**Any side-effects?**
Should resolve the build errors from the previous commit.

**Additional context/notes/links**

Resolves #139
